packages/blade/src/components/Dropdown/Dropdown.tsx

```diff
@@ -88,22 +90,28 @@ const _Dropdown = ({
   const isFirstRenderRef = React.useRef(true);
 
   React.useEffect(() => {
-    // Ignoring the `onDismiss` call on first render
+    // Ignoring the `onDismiss` and `onOpenChange` call on first render
     if (isFirstRenderRef.current) {
       isFirstRenderRef.current = false;
       return;
     }
 
-    if (!isOpen && onDismiss) {
-      onDismiss();
+    onOpenChange?.(isOpen);
+    if (!isOpen) {
+      onDismiss?.();
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isOpen]);
```

Comment:
We can use the onDidUpdate hook instead?

Reply:
aha nice! didn't know about this hook. Added